mesheryctl/internal/cli/root/pattern/onboard.go

```diff
@@ -203,8 +201,65 @@ func multipleApplicationsConfirmation(profiles []models.MesheryPattern) int {
 	}
 }
 
+func getSourceTypes() error {
+	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+	if err != nil {
+		utils.Log.Error(err)
+		return nil
+	}
+	validTypesURL := mctlCfg.GetBaseMesheryURL() + "/api/pattern/types"
+	req, err := utils.NewRequest("GET", validTypesURL, nil)
+	if err != nil {
+		utils.Log.Error(err)
+		return nil
+	}
+
+	resp, err := utils.MakeRequest(req)
+	if err != nil {
+		utils.Log.Error(err)
+		return nil
+	}
+
+	defer resp.Body.Close()
+
+	var response []*models.PatternSourceTypesAPIResponse
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		utils.Log.Error(utils.ErrReadResponseBody(errors.Wrap(err, "couldn't read response from server. Please try again after some time")))
+		return nil
+	}
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		utils.Log.Error(utils.ErrUnmarshal(errors.Wrap(err, "couldn't process response received from server")))
+		return nil
+	}
+
+	for _, apiResponse := range response {
+		validSourceTypes = append(validSourceTypes, apiResponse.DesignType)
+	}
+
+	return nil
+}
+
+// returns full source name e.g. helm -> `Helm Chart`
+// user passes only helm, compose or manifest but server accepts full source type
+// e.g `Heml Chart`, `Docker Compose`, `Kubernetes Manifest`
+func getFullSourceType(sType string) (string, error) {
+	for _, validType := range validSourceTypes {
+		lowerType := strings.ToLower(validType)
+		// user may pass Pascal Case source e.g Helm
+		sType = strings.ToLower(sType)
+		if strings.Contains(lowerType, sType) {
+			return validType, nil
+		}
+	}
+
+	return sType, fmt.Errorf("no matching source type found")
```

Comment:
Meshkit error

Reply:
Where's the meshkit error? Don't resolve reviews if they haven't been reached to conclusion.